### PR TITLE
Combat Correspondent

### PR DESCRIPTION
**Pull request by morrowwolf**
# About the pull request

This adds the role combat reporter. This role will be a primarily RP role. The role has an office south of shipside tcomms and west of laundry.

For the time being they have the same restrictions as the CL:
They may deploy at will.
They report directly to the acting commander with regards to ML.
They are NOT to be fighting xenos or frontlining. They should be staying in safe areas or moving with escorts (making the area safe).

# Explain why it's good for the game

This is something that has been asked for quite a bit. I'm on the fence about it personally but was requested to get something running.

There is a good argument against "role bloat" which is generally misunderstood as "too many roles bad" which is not the case. Instead, role bloat involves creating extra roles that take away from the gameplay loop or takes away from other roles (specifically in this case the corporate liaison who is rather similar if they want to do a gimmick).

Anyways, we'll see how the testmerge goes, feedback, as well as other architect thoughts.

Oh and the feedback word today is GIGGLES.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>
Tested a bit, we'll run into bugs I'm sure.
</details>

# Changelog

:cl: Morrow
add: Added the role Combat Correspondent.
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/radio/headset.dm [hunk 1]
```diff
@@ -511,6 +511,12 @@
 	icon_state = "wy_headset"
 	initial_keys = list(/obj/item/device/encryptionkey/mcom/cl)
 
+/obj/item/device/radio/headset/almayer/reporter
+	name = "reporter radio headset"
+	desc = "Used by the CL to convince people to sign NDAs. Channels are as follows: :v - marine command, :a - alpha squad, :b - bravo squad, :c - charlie squad, :d - delta squad, :n - engineering, :m - medbay, :u - requisitions, :j - JTAC, :t - intel, :y for WY."
```
Comment (BadAtThisGame302) [inline code/game/objects/items/devices/radio/headset.dm:516]: Change up the description or something?

**Comment by RoboticVodka** [comment]
As I'm aware this has been attempted multiple times one for an example(https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/2822) mainly denied because it was Civilian only and the proposed solution was to allow it to be both Civilian or Military with a toggle but never panned out. I'd also suggest you change the title of the job to Combat Correspondent, which is its real life equivalent 4341 https://en.wikipedia.org/wiki/List_of_United_States_Marine_Corps_MOS

**Comment by Nanu308** [comment]
> As I'm aware this has been attempted multiple times one for an example(https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/2822) mainly denied because it was Civilian only and the proposed solution was to allow it to be both Civilian or Military with a toggle but never panned out. I'd also suggest you change the title of the job to Combat Correspondent, which is its real life equivalent 4341 https://en.wikipedia.org/wiki/List_of_United_States_Marine_Corps_MOS

See this comment, same as prior issue.

**Comment by Mark9013100** [comment]
the press cap has no sprite when flipped

**Review by Agameofscones — CHANGES_REQUESTED** [review]
sprites check out fine (frans my beloved)

morrow my boy, integrate the office better. Don't stick it behind a maintenance door, remove that funny strip sitting in the middle of the way for reasons, and give the doors proper names.

**Comment by morrowwolf** [comment]
Mapping good to review.

Working on code changes still.

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 6a2cf07b91be24fbcbc651e8a46a8ae6445dd924
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 0f1dac4dcb22794ac0797cee8eb9c5b0b7c5e7f8

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: a6615543269deafdf5ad1f04caa9db43ed653967
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 566e2a7569e3c8fe4e83ec05f9b162d1c50e20d2

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: a6615543269deafdf5ad1f04caa9db43ed653967
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: e03f29db11392a076bad2f02448b2f3a3ac49415

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: a6615543269deafdf5ad1f04caa9db43ed653967
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 542ea8b549aa5ebcdd25d522cabb8346d1e8d294

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: a6615543269deafdf5ad1f04caa9db43ed653967
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 8788b2e03f66def7fd5673bae9a940858fd43b6b

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Removed

**Comment by cm13-github** [comment]
#### Test Merge Deployed

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: afc0a21cd466feb1aea83571d32fb955ff96ea5a
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: db68a422228a4ee3145ddefbd31174eac03b2d20

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 84c584f8a9e7f72c9dbf905625b4ef11866fe4ed
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 3d3051ddf73fc40c1c69a3dcbb0a2ca63ce0d550

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: f96439dea585696374f0f74eeaac0189748efd83
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 409c13271a4f43dc98037fb6b72ac4c64fd2eab8

##### Comment

</details>

**Comment by cm13live** [comment]
This pull request has been mentioned on **CM-SS13**. There might be relevant details there:

https://forum.cm-ss13.com/t/2761-combat-correspondent/811/1

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: f96439dea585696374f0f74eeaac0189748efd83
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 31dd7bac120691339519407fbaebb60be862f3c1

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: f96439dea585696374f0f74eeaac0189748efd83
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: c331f6e0c309e380bfe1c0b92001ed78b994ddcf

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: c121ee6752a4af8076dbce9aa63651c5802b098a
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: b951f3dbe2c5b84502bedc2316789f1b3ebf9abd

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: cb2716a6f3a07dfcd234d64d21a54d46e6955408
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 30b545f159c2cd382c9a9d84d80535401d456608

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: cb2716a6f3a07dfcd234d64d21a54d46e6955408
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 2a28b4aba52c786f6b9a56978af6f6e1ec5d6867

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 42bde1e0ccfeca19fcfcdd996a07e7ad42e1d623
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: db03b5326e26106c8c582b792adbd765dad1dd0f

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 46e938a9a59b7cf7b7f80392f1de3a6c3ed1f015
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 9ff946aa5d61af4999e86d21ab1ab48e99fa9c34

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: d61ac6d90c1ba3437c53b309b80e628098ae2c16
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 15a5e154b257e998812c319dc1801712110330b1

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: d010b1ac6502581f9c54fd5d9ab1e04bd72d2534
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 58891f7ef62fd1ba5f776bea224bcd2f4ed37b2a

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 50f3bad1a51f71bf2c5d29e8236f663d2ca65441
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 965ce05a44359009a22aa3c23fdab9d57af3556f

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 24ced9126768119f480e6a8d6d91ac4ba8f25119
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 1d8fe425cd8dc8ec96a88e84fa97dfb04c7f6c47

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 24ced9126768119f480e6a8d6d91ac4ba8f25119
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 49d1b8c0ff4d15efdbd4fd0cbecd571fb02623ee

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: a712c10248439405f450f07ca7c7acf69bb848d7
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: fff4ba6badffbafa4be46c98823ac0c89df423b6

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 58a51e1246fd072d444ab6bf09944be5f02b92d6
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: 34f1cd3bb642446c355fb98cd2cf7255899d2fed

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: d817dc374787ac3dcce4d37d7b5aaa6bf45aba9b
Pull Request: e2d5f5d1bcdecf9f89bc650099d15ec1684dfa34
Server: eba44f3e93548d2616501d9486bc358c491fdf14

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 4708853a780fe8dc72848aaef4a440011ac4490f
Pull Request: df0d065030e1adfb6d52c6d78d96761a2d821d98
Server: 00a5ba229c15728cc98656fec14ae59ca56c401b

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 4708853a780fe8dc72848aaef4a440011ac4490f
Pull Request: df0d065030e1adfb6d52c6d78d96761a2d821d98
Server: 2553f9eae54e1e8fba39250f1fbaef3aad25d1af

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 4708853a780fe8dc72848aaef4a440011ac4490f
Pull Request: df0d065030e1adfb6d52c6d78d96761a2d821d98
Server: cdd6fe2764fa15eae4632c12aeee0e9d5ece932e

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 40e1c33dd3db0c09405cfebc5c063f0bab3afab7
Pull Request: df0d065030e1adfb6d52c6d78d96761a2d821d98
Server: 51152bdccda4fb695b4f0e8f74f0a6ea3b01a813

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 40e1c33dd3db0c09405cfebc5c063f0bab3afab7
Pull Request: df0d065030e1adfb6d52c6d78d96761a2d821d98
Server: cddc03115ac4de96c00a534facfcfa14c9ba6b07

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 0dea5e168383a7e3de233f2e66e92049f13e139c
Pull Request: df0d065030e1adfb6d52c6d78d96761a2d821d98
Server: 7574a7f61af3be4dbaec4e7bb5f15e7a5932b17f

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 0dea5e168383a7e3de233f2e66e92049f13e139c
Pull Request: a83a784250d92ddde1b2af3050828c539bd14041
Server: 6bb9e7d49ab844c93c1c645dfa57f2ce22e924b2

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 0dea5e168383a7e3de233f2e66e92049f13e139c
Pull Request: 019e2401ad930505435cc90cc06d995ba16fc252
Server: 2f52ba8deba310a1f8ff051abba991e66698653d

##### Comment

</details>

File: code/modules/gear_presets/uscm_ship.dm [hunk 2]
```diff
@@ -98,6 +98,92 @@
...
+	paygrade = "ME4"
+	role_comm_title = "CC"
+	skills = /datum/skills/pfc
+	idtype = /obj/item/card/id/dogtag
+
+	minimap_icon = "correspondent"
+	minimap_background = MINIMAP_ICON_BACKGROUND_CIC
+
+/datum/equipment_preset/uscm_ship/reporter_uscm/load_gear(mob/living/carbon/human/H)
+	var/backItem = /obj/item/storage/backpack/satchel
```
Comment (harryob) [inline code/modules/gear_presets/uscm_ship.dm:162]: snake case

File: code/modules/gear_presets/uscm_ship.dm [hunk 3]
```diff
@@ -98,6 +98,92 @@
...
+	minimap_background = MINIMAP_ICON_BACKGROUND_CIVILIAN
+
+/datum/equipment_preset/uscm_ship/reporter/load_gear(mob/living/carbon/human/H)
+	var/backItem = /obj/item/storage/backpack/satchel
+
+	H.equip_to_slot_or_del(new /obj/item/device/radio/headset/almayer/reporter(H), WEAR_L_EAR)
+	H.equip_to_slot_or_del(new /obj/item/clothing/under/marine/reporter(H), WEAR_BODY)
+	H.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/jacket/marine/reporter(H), WEAR_JACKET)
+	H.equip_to_slot_or_del(new /obj/item/clothing/shoes/laceup(H), WEAR_FEET)
+	H.equip_to_slot_or_del(new backItem(H), WEAR_BACK)
```
Comment (harryob) [inline code/modules/gear_presets/uscm_ship.dm:136]: snakecase

File: code/modules/gear_presets/uscm_ship.dm [hunk 4]
```diff
@@ -98,6 +98,92 @@
...
+	paygrade = "C"
+	role_comm_title = "PRESS"
+	skills = /datum/skills/civilian
+	idtype = /obj/item/card/id/silver/cl
+
+	minimap_icon = "correspondent"
+	minimap_background = MINIMAP_ICON_BACKGROUND_CIVILIAN
+
+/datum/equipment_preset/uscm_ship/reporter/load_gear(mob/living/carbon/human/H)
+	var/backItem = /obj/item/storage/backpack/satchel
```
Comment (harryob) [inline code/modules/gear_presets/uscm_ship.dm:130]: snakecase

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: e683278fb1d596c51e80f921a1810cbad53324db
Pull Request: 019e2401ad930505435cc90cc06d995ba16fc252
Server: 81966dcdc7a3f248e4a56037c647d30c9fcd8500

##### Comment

</details>

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: e683278fb1d596c51e80f921a1810cbad53324db
Pull Request: 019e2401ad930505435cc90cc06d995ba16fc252
Server: 7a8d2996afeaaa238369233ecd3ee0aa0b4ec146

##### Comment

</details>

File: code/modules/admin/topic/topic.dm
Comment (harryob) [inline code/modules/admin/topic/topic.dm:1]: i'm so unhappy

File: code/modules/paperwork/notepad.dm [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+			else if(page == paper_left)
+				screen = 1
+			page--
+			playsound(loc, "pageturn", 15, 1)
+		if(href_list["remove"])
+			if(contents.len < page)
+				page = contents.len
+			var/obj/item/ripped_out_page = contents[page]
+			usr.put_in_hands(ripped_out_page)
+			to_chat(usr, SPAN_NOTICE("You rip out \the [ripped_out_page.name] from \the [src]."))
```
Comment (harryob) [inline code/modules/paperwork/notepad.dm:98]:
```suggestion
			to_chat(usr, SPAN_NOTICE("You rip out [ripped_out_page] from [src]."))
```

File: code/modules/paperwork/notepad.dm [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+		to_chat(usr, SPAN_NOTICE("Ghosts don't have hands, you can't flip the page!"))
+	else
+		to_chat(usr, SPAN_NOTICE("You need to hold it in your hands!"))
+
+/obj/item/notepad/verb/rename()
+	set name = "Rename notepad"
+	set category = "Object"
+	set src in usr
+
+	var/n_name = strip_html(input(usr, "What would you like to label the notepad?", "notepad Labelling", null)  as text)
```
Comment (harryob) [inline code/modules/paperwork/notepad.dm:124]: use tgui_input_text and get rid of the as text and stuff

File: code/modules/paperwork/notepad.dm [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+	if((loc == usr && usr.stat == 0))
```
Comment (harryob) [inline code/modules/paperwork/notepad.dm:125]: stat has defines

File: code/modules/admin/topic/topic.dm [hunk 8]
```diff
@@ -1265,6 +1265,88 @@
...
+		GLOB.fax_contents += fax_message // save a copy
+
+		GLOB.USCMFaxes.Add("<a href='?FaxView=\ref[fax_message]'>\[view reply at [world.timeofday]\]</a>")
+
+		var/customname = input(src.owner, "Pick a title for the report", "Title") as text|null
+
+		var/msg_ghost = SPAN_NOTICE("<b><font color='#1F66A0'>PRESS REPLY: </font></b> ")
+		msg_ghost += "Transmitting '[customname]' via secure connection ... "
+		msg_ghost += "<a href='?FaxView=\ref[fax_message]'>view message</a>"
+		announce_fax( ,msg_ghost)
```
Comment (harryob) [inline code/modules/admin/topic/topic.dm:1319]: named arguments

File: code/game/jobs/job/civilians/other/reporter.dm [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+	spawn_positions = 1
+	selection_class = "job_cl"
+	supervisors = "the acting commanding officer"
+	gear_preset = /datum/equipment_preset/uscm_ship/reporter
+	flags_startup_parameters = ROLE_ADD_TO_DEFAULT
+	selection_class = "job_cl"
+
+	job_options = list(CIVILIAN_VARIANT, MILITARY_VARIANT)
+
+	var/military = FALSE
```
Comment (harryob) [inline code/game/jobs/job/civilians/other/reporter.dm:16]:
```suggestion
	/// If this job is a military variant of the reporter role
	var/military = FALSE
```
or something to that effect

File: code/game/jobs/job/civilians/other/reporter.dm [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+/datum/job/civilian/reporter/handle_job_options(option)
+	if(option != job_options[1])
```
Comment (harryob) [inline code/game/jobs/job/civilians/other/reporter.dm:19]: hard coding based on the position in the list isn't ideal, if option == CIVILIAN_VARIANT would be fine (or associating the list of job options with gear presets or w/e)

File: code/game/jobs/job/job.dm [hunk 11]
```diff
@@ -300,3 +303,6 @@
 		SSround_recording.recorder.track_player(human)
 
 	return TRUE
+
+/datum/job/proc/handle_job_options(option)
+	return
```
Comment (harryob) [inline code/game/jobs/job/job.dm:308]:
```suggestion
/// Intended to be overwritten to handle when a job has variants that can be selected.
/datum/job/proc/handle_job_options(option)
	return
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Pull request merged**